src/BSON/UTCDateTime.c

```diff
@@ -161,49 +157,58 @@ static bool php_phongo_utcdatetime_init_from_date(php_phongo_utcdatetime_t *inte
 	return true;
 }
 
-/* {{{ proto void UTCDateTime::__construct([string|DateTimeInterface $milliseconds = null])
+/* {{{ proto void UTCDateTime::__construct([int|float|string|DateTimeInterface $milliseconds = null])
    Construct a new BSON UTCDateTime type from either the current time,
-   milliseconds since the epoch, or a DateTimeInterface object. */
+   milliseconds since the epoch, or a DateTimeInterface object. Defaults to the
+   current time. */
 PHP_METHOD(UTCDateTime, __construct)
 {
 	php_phongo_utcdatetime_t    *intern;
 	zend_error_handling          error_handling;
-	zval                        *datetime = NULL;
+	zval                        *milliseconds = NULL;
 
 	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
 	intern = Z_UTCDATETIME_OBJ_P(getThis());
 
-	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "|o!", &datetime) == SUCCESS) {
-		if (datetime == NULL) {
-			php_phongo_utcdatetime_init_from_current_time(intern);
-		} else if (instanceof_function(Z_OBJCE_P(datetime), php_date_get_date_ce() TSRMLS_CC)) {
-			php_phongo_utcdatetime_init_from_date(intern, Z_PHPDATE_P(datetime));
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|z!", &milliseconds) == FAILURE) {
+		zend_restore_error_handling(&error_handling TSRMLS_CC);
+		return;
+	}
+	zend_restore_error_handling(&error_handling TSRMLS_CC);
+
+	if (milliseconds == NULL) {
+		php_phongo_utcdatetime_init_from_current_time(intern);
+		return;
+	}
+
+	if (Z_TYPE_P(milliseconds) == IS_OBJECT) {
+		if (instanceof_function(Z_OBJCE_P(milliseconds), php_date_get_date_ce() TSRMLS_CC)) {
+			php_phongo_utcdatetime_init_from_date(intern, Z_PHPDATE_P(milliseconds));
 #if PHP_VERSION_ID >= 50500
-		} else if (instanceof_function(Z_OBJCE_P(datetime), php_date_get_immutable_ce() TSRMLS_CC)) {
-			php_phongo_utcdatetime_init_from_date(intern, Z_PHPDATE_P(datetime));
+		} else if (instanceof_function(Z_OBJCE_P(milliseconds), php_date_get_immutable_ce() TSRMLS_CC)) {
+			php_phongo_utcdatetime_init_from_date(intern, Z_PHPDATE_P(milliseconds));
 #endif
 		} else {
-			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected instance of DateTimeInterface, %s given", ZSTR_VAL(Z_OBJCE_P(datetime)->name));
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected instance of DateTimeInterface, %s given", ZSTR_VAL(Z_OBJCE_P(milliseconds)->name));
 		}
-
-		zend_restore_error_handling(&error_handling TSRMLS_CC);
 		return;
 	}
 
-	{
-		char                *s_milliseconds;
-		phongo_zpp_char_len  s_milliseconds_len;
-
-		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &s_milliseconds, &s_milliseconds_len) == FAILURE) {
-			zend_restore_error_handling(&error_handling TSRMLS_CC);
-			return;
-		}
+	if (Z_TYPE_P(milliseconds) == IS_DOUBLE) {
```

Comment:
I admittedly has some trouble reading through all of the `zend_dval_to_lval()` cases to see if this would be an issue. Would you agree that the best solution is to convert this to a string via `"%f.0"`?